scripts/ci/logcheck/allowlist-patterns

```diff
@@ -20,3 +20,7 @@ unexpected owner for certificate secrets
 # We detect if they are not supported and in that case do not apply them, so these warnings are not useful at this point.
 # See ROX-17796 ROX-17766 ROX-17734 for related work.
 policy/v1beta1 PodSecurityPolicy is deprecated in v1.21., unavailable in v1.25.
+# This prevents false-positives whin testing on OCP 4.14.
+# DeploymentConfigs are deprecated but there are no plans to remove them as this is written.
+# See: https://issues.redhat.com/browse/OCPSTRAT-118
+apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
```

Comment:
The problem is that lines in this file should be regular expressions. Besides interpreting `.` as any character, `/` should probably be escaped, as well as `+`.

Regarding testing: you don't need to run this through 4.14 cluster, you can run `scripts/ci/logcheck/check.sh` standalone against some hand-crafted input.